src/Documents/Document.php

```diff
@@ -10,14 +10,19 @@ final class Document implements ArrayableInterface
      * @var string
      */
     private $id;
+    /**
+     * @var string|null
+     */
+    private $routing;
     /**
      * @var array
      */
     private $content;
 
-    public function __construct(string $id, array $content)
+    public function __construct(string $id, array $content, ?string $routing = null)
```

Comment:
Personally, I don't like this idea as routing doesn't belong to the document. It just controls where the document is stored inside the index.

I'd prefer to extend `DocumentManager:: index()`/`DocumentManager:: delete()`  as follows:

```php
public function index(string $indexName, array $documents, bool $refresh = false, string $routingPath = null): self
{
     // $routingPath represents a document field, which is used for routing, 
     // it's similar to deprecated mapping _routing.path
     // ..
     $index = ['_id' => $document->getId()];

     if (isset($routingPath)) {
          $index['routing'] = $document->getField($routingPath);
     }
     
     $params['body'][] = ['index' => $index];
     $params['body'][] = $document->getContent();
     // ..
}
```

Reply:
@babenkoivan I see your point but the problem is that routing is document specific and we need a way to define different routing per document. So the bulk operation could not work as you suggest because then that would imply that all given documents are routed to the same shard.

Also from es [docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-routing-field.html) we can see that they state the following:
_Custom routing patterns can be implemented by specifying a custom routing value **per document**_

Reply:
Hmm i see now that you intended for routingPath to be a field and not an actual value. I guess that could work but only for index operations and not deletes. Also it limits routing to only fields that exist inside the model which should normally be ok but it would be a limitation of the library and not elasticsearch.

Reply:
Will try your suggested implementation , i guess the above issue i mention shouldn't be a real problem for anyone and can be fixed by adding another field in the document data just for routing, in case a developer really needs it.